modules/questions.py

```diff
@@ -32,226 +27,418 @@
+
+    def get_question_by_id(self, row_id: str) -> Optional[ParsedRow]:
+        """Get question by id in "All Answers" table.
+        Returns `ParsedRow` or `None` (if question with that id doesn't exist)
+        """
+        uri = f"https://coda.io/apis/v1/docs/{self.DOC}/tables/{self.ALL_ANSWERS_TABLE}/rows/{row_id}"
+        params = {
+            "valueFormat": "simple",
+            "useColumnNames": True,
+        }
+        req = requests.get(uri, params=params, headers=self.get_headers())
+        if req.status_code == 200:
+            return parse_row(req.json())
+
+    def get_question_by_gdoc_link(self, url: str) -> Optional[ParsedRow]:
+        """Get question by link to its GDoc.
+        Returns `ParsedRow` or `None` (if question with that id doesn't exist)
+        """
+        questions = self.get_questions_df()
+        questions_queried = questions[questions["url"].str.startswith(url)]
+        if not questions_queried.empty:
+            return cast(ParsedRow, questions_queried.iloc[0].to_dict())
+
     @classmethod
     def get_status_shorthand_dict(cls) -> dict[str, str]:
-        """Get dictionary mapping statuses and shorthands for statuses
-        to actual status labels as appear in coda
+        """Get dictionary mapping statuses and status shorthands
+        (e.g. "bs" for "Bulletpoint sketch") to valid Status labels.
         """
         # Workaround to make mock request during testing
-        if cls.CODA_API_TOKEN == "testing":
+        if cls.is_in_testing_mode():
             return {}
 
-        response = cls.send_coda_questions_request()
-        status_vals = {row["values"]["Status"] for row in response["items"]}
-        shorthand_dict = {}
-        for status_val in status_vals:
-            shorthand_dict[status_val] = status_val
-            shorthand_dict[status_val.lower()] = status_val
-            shorthand = "".join(word[0].lower() for word in status_val.split())
-            shorthand_dict[shorthand] = status_val
-        return shorthand_dict
+        statuses = cls.get_all_statuses()
+        status_shorthand_dict = {}
+        for status in statuses:
+            status_shorthand_dict[status] = status
+            status_shorthand_dict[status.lower()] = status
+            shorthand = "".join(word[0].lower() for word in status.split())
+            status_shorthand_dict[shorthand] = status
+        return status_shorthand_dict
+
+    @classmethod
+    def get_all_tags(cls) -> list[str]:
+        """Get all tags from "All Answers" table"""
+        # Workaround to make mock request during testing
+        if cls.is_in_testing_mode():
+            return []
+
+        response = cls.send_all_answers_table_request()
+        all_tags: set[str] = set(  # TODO: make it more readable
+            filter(
+                bool,
+                reduce(
+                    add, [row["values"]["Tags"].split(",") for row in response["items"]]
+                ),
+            )
+        )
```

Comment:
not tested, but should work in theory:tm:
```suggestion
        all_tags: set[str] = set(row["values"]["Tags"].split(",") for row in response["items"] if row["values"]["Tags"])
```